src/persist/src/indexed/columnar.rs

```diff
@@ -98,6 +98,28 @@ impl<'a> FromIterator<&'a ((Vec<u8>, Vec<u8>), u64, isize)> for ColumnarRecords
     }
 }
 
+impl<'a> FromIterator<((&'a [u8], &'a [u8]), u64, isize)> for ColumnarRecords {
```

Comment:
seems like we could combine these two impls, but I'm happy leaving that as a TODO